src/components/containers/Panel.js

```diff
@@ -1,19 +1,46 @@
 import PanelHeader from './PanelHeader';
+import PanelEmpty from './PanelEmpty';
 import PropTypes from 'prop-types';
 import React, {Component, cloneElement} from 'react';
 import update from 'immutability-helper';
-import {bem} from 'lib';
+import {bem, localize} from 'lib';
+import {EmbedIconIcon} from 'plotly-icons';
+
+class PanelErrorImpl extends Component {
+  render() {
+    const {localize: _} = this.props;
+
+    return (
+      <PanelEmpty
+        icon={EmbedIconIcon}
+        heading={_('Well this is embarrassing.')}
+        message={_('This panel could not be displayed due to an error.')}
+      />
+    );
+  }
+}
+
+PanelErrorImpl.propTypes = {
+  localize: PropTypes.func,
+};
+
+const PanelError = localize(PanelErrorImpl);
 
 class Panel extends Component {
   constructor(props) {
     super(props);
     this.state = {
       individualFoldStates: [],
+      hasError: false,
     };
     this.toggleFolds = this.toggleFolds.bind(this);
     this.toggleFold = this.toggleFold.bind(this);
   }
 
+  componentDidCatch() {
+    this.setState({hasError: true});
+  }
```

Comment:
was there any more details we could have logged with console.error? componentDidCatch does take an error argument that maybe we could log?

Reply:
it doesn't log anything more than that screenshot I posted, just in a JSON object, so I didn't include it. It would/will be useful to log this kind of error to a production-monitoring system or something like that in the future. Like Errbit or Rollbar.